bitnami/opensearch/values.yaml

```diff
@@ -3408,3 +3408,181 @@ dashboards:
+  ## @param snapshots.containerSecurityContext.capabilities.drop List of capabilities to be dropped
+  ## @param snapshots.containerSecurityContext.seccompProfile.type Set container's Security Context seccomp profile
+  ##
+  containerSecurityContext:
+    enabled: true
+    seLinuxOptions: null
+    runAsUser: 1001
+    runAsGroup: 1001
+    runAsNonRoot: true
+    privileged: false
+    readOnlyRootFilesystem: true
+    allowPrivilegeEscalation: false
+    capabilities:
+      drop: ["ALL"]
+    seccompProfile:
+      type: "RuntimeDefault"
+  ## @param snapshots.enabled Enable automatic setup of repositories and snapshot policies
+  ##
+  enabled: false
+  ## @param snapshots.fullnameOverride String to fully override opensearch.snapshots.fullname
+  ##
+  fullnameOverride: ""
+  ## Bitnami OpenSearch Snapshots image
+  ## @param snapshots.image.registry OpenSearch Snapshots image registry
+  ## @param snapshots.image.repository OpenSearch Snapshots image repository
+  ## @skip snapshots.image.tag OpenSearch Snapshots image tag (immutable tags are recommended)
+  ## @param snapshots.image.digest OpenSearch Snapshots image digest in the way sha256:aa.... Please note this parameter, if set, will override the tag
+  ## @param snapshots.image.pullPolicy OpenSearch Snapshots image pull policy
+  ## @param snapshots.image.pullSecrets OpenSearch Snapshots image pull secrets
+  ## @param snapshots.image.debug Enable OpenSearch Snapshots image debug mode
+  ##
+  image:
+    registry: docker.io
+    repository: bitnami/os-shell
+    tag: 12-debian-12-r30
+    digest: ""
+    ## Specify a imagePullPolicy
+    ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
+    ## ref: http://kubernetes.io/docs/concepts/containers/images/#pre-pulled-images
+    ##
+    pullPolicy: IfNotPresent
+    ## Optionally specify an array of imagePullSecrets.
+    ## Secrets must be manually created in the namespace.
+    ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+    ## e.g:
+    ## pullSecrets:
+    ##   - myRegistryKeySecretName
+    ##
+    pullSecrets: []
+    ## Enable debug mode
+    ##
+    debug: false
+  ## @param snapshots.nameOverride String to partially override common.names.fullname
+  ##
+  nameOverride: ""
+  ## Enable persistence using Persistent Volume Claims
+  ## ref: https://kubernetes.io/docs/concepts/storage/persistent-volumes/
+  ##
+  persistence:
+    ## @param snapshots.persistence.accessModes Persistent Volume Access Modes
+    ##
+    accessModes:
+      - ReadWriteMany
+    ## @param snapshots.persistence.annotations Persistent Volume Claim annotations
+    ##
+    annotations: {}
+    ## @param snapshots.persistence.dataSource Custom PVC data source
+    ##
+    dataSource: {}
+    ## @param snapshots.persistence.enabled Enable persistence using Persistent Volume Claims
+    ##
+    enabled: false
```

Comment:
I'm not convinced that it makes sense to let users disable PVC for snapshots. What kind of use-case is that? I think it is a strange interdependence between `snapshots.enabled` and `snapshots.persistence.enabled` right now; only `snapshots.enabled` should be required, in my opinion. But keeping this makes it more similar to other config in the chart.

WDYT?

Reply:
I prefer keeping the "snapshots.persistence.enabled" parameter and set it to true by default to follow the same approach we have for other solutions. For example, we mount an emptyVolume if persistence is not configured in MySQL

```
  {{- if and .Values.primary.persistence.enabled .Values.primary.persistence.existingClaim }}
        - name: data
          persistentVolumeClaim:
            claimName: {{ tpl .Values.primary.persistence.existingClaim . }}
  {{- else if not .Values.primary.persistence.enabled }}
        - name: data
          emptyDir: {}
...
```

You can follow that approach here. This case covers the scenarios where you can't create/configure a volume and need to disable persistence.

Reply:
> This case covers the scenarios where you can't create/configure a volume and need to disable persistence.

I don't see any value for them to have `snapshots.enabled=true` unless they also have persistent volumes. But whatever – I'm happy to do what you suggest here and keep the config option as already configured.

Reply:
>Some components have persistence.enabled=true by default, but not all. At least dashboards.persistence.enabled=false by default. So there is no common approach to follow here.

As you are configuring an "extra" feature, I'd disable the persistence by default. Users can always enable it when deploying the solution

>We must not set snapshots.persistence.enabled=true by default unless we also set snapshots.enabled=true by default also. Otherwise, we'll just mount volumes that are never used, which makes no sense.

That's correct. 

>I suggest we remove the snapshots.persistence.enabled

I don't like this idea. I prefer providing the users with the possibility to manage this information. 

> or snapshots.enabled=false,snapshots.persistence.enabled=false by default.

This is the approach I'd follow. Sounds good?

Reply:
Sounds all good. Thanks for the input. 👍